libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -49,6 +49,35 @@ using namespace solidity::frontend;
 using namespace solidity::langutil;
 using namespace solidity::util;
 
+namespace
+{
+
+Type const* closestType(Type const* _type, Type const* _targetType, bool _isShiftOp)
+{
+	if (_isShiftOp)
+		return _type->mobileType();
+	else if (_type->category() != Type::Category::Tuple)
```

Comment:
Can you rearrange the cases so that there is no negation? `else if (auto tupleType = dynamic_cast<...>)` could also be shorter.

Reply:
Done